### PR TITLE
CRDCDH-1614 Misc. CLI Config File dialog updates

### PR DESCRIPTION
**Pull request by amattu2**
### Overview

This PR modifies the CLI Configuration download dialog to make fields required and update the verbiage to indicate S3 buckets are now supported.

### Change Details (Specifics)

- Update the CLI Config form to make the data file folder path and file manifest path fields required
- Update the CLI Config dialog to memoize it
- Standardize the dataType logic in the DataUpload component
- Update test coverage for the misc. changes

### Related Ticket(s)

CRDCDH-1614 (Task)
CRDCDH-1565 (User Story)

### CODE REVIEW

File: src/components/UploaderConfigDialog/index.tsx [hunk 1]
```diff
@@ -151,33 +156,50 @@ const UploaderConfigDialog: FC<Props> = ({ onClose, onDownload, open, ...rest })
...
         </StyledBodyText>
-        <StyledForm onSubmit={handleDownload}>
+        <StyledForm onSubmit={handleSubmit(onSubmit)} id="uploader-config-dialog-form">
           <Box>
-            <StyledLabel id="data-folder-input-label">Local Path of Data Files Folder</StyledLabel>
+            <StyledLabel id="data-folder-input-label">
+              Full Path to Data Files Folder
+              <StyledAsterisk />
+              <Tooltip
+                title="Enter the full path for the Data Files folder on your local machine or S3 bucket"
```
Comment (Alejandro-Vega) [inline src/components/UploaderConfigDialog/index.tsx:171]: All other tooltips in with a period, maybe we should consider adding it here as well.

File: src/components/UploaderConfigDialog/index.tsx [hunk 2]
```diff
@@ -151,33 +156,50 @@ const UploaderConfigDialog: FC<Props> = ({ onClose, onDownload, open, ...rest })
...
             />
+            <StyledHelperText>{errors?.dataFolder?.message}</StyledHelperText>
           </Box>
           <Box>
-            <StyledLabel id="manifest-input-label">Local Path of Manifest File</StyledLabel>
+            <StyledLabel id="manifest-input-label">
+              Full Path to Manifest File
+              <StyledAsterisk />
+              <Tooltip
+                title="Enter the full path for the File Manifest on your local machine or S3 bucket"
```
Comment (Alejandro-Vega) [inline src/components/UploaderConfigDialog/index.tsx:190]: All other tooltips in with a period, maybe we should consider adding it here as well.

File: src/components/UploaderConfigDialog/index.tsx [hunk 3]
```diff
@@ -151,33 +156,50 @@ const UploaderConfigDialog: FC<Props> = ({ onClose, onDownload, open, ...rest })
...
+              <Tooltip
+                title="Enter the full path for the Data Files folder on your local machine or S3 bucket"
+                open={undefined}
+                disableHoverListener={false}
+                data-testid="data-folder-input-tooltip"
+              />
+            </StyledLabel>
             <StyledOutlinedInput
-              {...register("dataFolder", { required: true })}
+              {...register("dataFolder", { required: "This field is required" })}
```
[the remark this answers is not in the document]
Comment (amattu2) [reply src/components/UploaderConfigDialog/index.tsx]: Completely agree. Thanks for checking this. I updated the implementation and also test coverage for these scenarios. 3e386a8

File: src/components/UploaderConfigDialog/index.tsx (hunk 1 above)
Comment (amattu2) [reply src/components/UploaderConfigDialog/index.tsx]: Agreed. I updated the reqs to add periods and updated the implementation to match. 3e386a8

**Review by Alejandro-Vega — APPROVED** [review]
LGTM!

**Pull request merged**